### PR TITLE
Fixed edge cases for Trig/Log/Exp XMath functions

### PR DESCRIPTION
**Pull request by MoFtZ**
Fixes some of the issues with https://github.com/m4rs-mt/ILGPU.Algorithms/issues/16
NOTE: the `Pow` function is still not fixed, and does not deal with negative 'base'. This is a known limitation of the way we calculate `Pow`.

Requires https://github.com/m4rs-mt/ILGPU/pull/91 and https://github.com/m4rs-mt/ILGPU/pull/92

### CODE REVIEW

File: Src/ILGPU.Algorithms.Tests/XMathTests.cs [hunk 1]
```diff
@@ -35,8 +35,17 @@ private class FloatPrecisionComparer
...
 
-            public override bool Equals(float x, float y) =>
-                Math.Abs(x - y) < Margin;
+            public override bool Equals(float x, float y)
+            {
+                if ((float.IsNaN(x) && float.IsNaN(y)) ||
+                    (float.IsPositiveInfinity(x) && float.IsPositiveInfinity(y)) ||
+                    (float.IsNegativeInfinity(x) && float.IsNegativeInfinity(y)))
+                {
+                    return true;
```
Comment (m4rs-mt) [inline Src/ILGPU.Algorithms.Tests/XMathTests.cs:44]: What happen in cases in which `x = +Infinity` and `y = -Infinity`? Does this ever happen? If yes, is this case important to capture?

**Review by m4rs-mt — APPROVED** [review]

File: Src/ILGPU.Algorithms.Tests/XMathTests.cs (hunk 1 above)
Comment (MoFtZ) [reply Src/ILGPU.Algorithms.Tests/XMathTests.cs:44]: It's not immediately obvious, even to me, so I'll fix it up to be explicit.

When you subtract from opposite infinities, it does not change:
`+Infinity` - `-Infinity` = `+Infinity`
`-Infinity` - `+Infinity` = `-Infinity`

When you add opposite infinities, it changes:
`+Infinity` + `-Infinity` = `NaN`
`-Infinity` + `+Infinity` = `NaN`

We only use subtraction in the comparer, so the current code just happens to work...

**Comment by MoFtZ** [comment]
Added explicit checks for comparing `+Infinity` to `-Infinity`.

Added edge cases to unit tests for Rcp, Sqrt, Rsqrt, Exp and Exp2.

File: Src/ILGPU.Algorithms.Tests/XMathTests.cs (hunk 1 above)
Comment (m4rs-mt) [reply Src/ILGPU.Algorithms.Tests/XMathTests.cs:44]: Yeah, you're right. It is always confusing when arguing about 'NaN's' and 'Infinity' float values 🔢

**Review by m4rs-mt — APPROVED** [review]
